pkg/helper/loki_config.go

```diff
@@ -31,15 +31,15 @@ func NewLokiConfig(spec *flowslatest.FlowCollectorLoki) LokiConfig {
 	}
 	switch spec.Mode {
 	case flowslatest.LokiModeLokiStack:
-		dotNamespace := ""
+		ns := namespace
 		if len(spec.LokiStack.Namespace) > 0 {
-			dotNamespace = "." + spec.LokiStack.Namespace
+			ns = spec.LokiStack.Namespace
 		}
-		gatewayURL := fmt.Sprintf("https://%s-gateway-http%s.svc:8080/api/logs/v1/network/", spec.LokiStack.Name, dotNamespace)
+		gatewayURL := fmt.Sprintf("https://%s-gateway-http.%s.svc:8080/api/logs/v1/network/", spec.LokiStack.Name, ns)
 		loki.LokiManualParams = flowslatest.LokiManualParams{
 			QuerierURL:  gatewayURL,
 			IngesterURL: gatewayURL,
-			StatusURL:   fmt.Sprintf("https://%s-query-frontend-http%s.svc:3100/", spec.LokiStack.Name, dotNamespace),
+			StatusURL:   fmt.Sprintf("https://%s-query-frontend-http.%s.svc:3100/", spec.LokiStack.Name, ns),
```

Comment:
when ns is `""` there is no need to have `.` in the url name right ?

Reply:
When namespace is "", the URL won't be correct, no matter if there's the `.` or not, this is precisely the issue that this PR is fixing. So here, an empty namespace will still result in an invalid URL but it is ultimately replaced with the netobserv's namespace which fixes it all.